### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by dunkgray**
Pulling out more stand-alone functions in preparation for ls9 reprocessing.
Plus fixing tests

### CODE REVIEW

File: tests/test_do_ard.py [hunk 1]
```diff
@@ -24,24 +68,32 @@ def test_do_ard(tmp_path):
...
+    )
 
     # Not checking the values, just that the files exist
-    assert os.path.exists(tmp_path / "jobdir" / ARCHIVE_FILE), ARCHIVE_FILE + ' does not exist.'
-    assert os.path.exists(tmp_path / "jobdir" / ODC_FILTERED_FILE), ODC_FILTERED_FILE + ' does not exist.'
-    assert os.path.exists(tmp_path / "jobdir" / PBS_ARD_FILE), PBS_ARD_FILE + ' does not exist.'
+    assert os.path.exists(tmp_path / "jobdir" / ARCHIVE_FILE), (
+        ARCHIVE_FILE + " does not exist."
+    )
+    assert os.path.exists(tmp_path / "jobdir" / ODC_FILTERED_FILE), (
```
Comment (gordonyeong) [inline tests/test_do_ard.py:94]: Recommendation: I think it's safer to join the path elements using os.path.join()

File: tests/test_utils.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+def test_local_path():
+    # s2
+    # uris = ["zip:/yada/yada/yada20124T021536.zip!/"]
+
+    # ls
+    # "local_path": "PosixPath('/g/data/u46/yada_01_T2.odc-metadata.yaml')
+    s2_l1_dataset = Mock()
+    s2_l1_dataset.local_path = None
+    path = "/g/S2A_MSIL1C_20220124T004711_N0301_R102_T54LYH_20220124T021536.zip"
+    s2_l1_dataset.uris = ["zip:" + path + "!/"]
```
Comment (gordonyeong) [inline tests/test_utils.py:17]: recommendation: might be safer to concat the uri path elements using urllib.parse.urljoin()

File: tests/test_utils.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    product_id = "S2A_OPER_MSI_L1C_TL_VGS2_20220124T021536_A034419_T54LYH_N03.01"
+    result = calc_file_path(s2_l1_dataset, product_id)
+    assert result == path
+
+    ls_l1_dataset = Mock()
+    the_path = "/this/path/"
+    ls_l1_dataset.local_path = Path(the_path + "LC08_T2.odc-metadata.yaml")
```
Comment (gordonyeong) [inline tests/test_utils.py:24]: Recommend: [join()](https://docs.python.org/3/library/os.path.html#os.path.join)

File: tests/test_utils.py (hunk 2 above)
Comment (dunkgray) [reply tests/test_utils.py:18]: Thanks. Done.

File: tests/test_do_ard.py (hunk 1 above)
Comment (dunkgray) [reply tests/test_do_ard.py:94]: This is ok for Path objects.

File: tests/test_utils.py (hunk 3 above)
Comment (dunkgray) [reply tests/test_utils.py]: Thanks. Done

File: tests/test_utils.py (hunk 2 above)
Comment (dunkgray) [reply tests/test_utils.py:18]: This is more testing the string, so urllib.parse.urljoin() wouldn't work.

**Review by gordonyeong — APPROVED** [review]

**Pull request merged**